clusterloader2/pkg/measurement/common/network/network_performance_measurement.go

```diff
@@ -0,0 +1,449 @@
+		UpdateFunc: func(oldObj interface{}, newObj interface{}) {
+			npm.handleUpdateNetworkTestEvents(newObj)
+		},
+	})
+	npm.stopCh = make(chan struct{})
+	go informer.Run(npm.stopCh)
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) prepareCluster() error {
+	if err := client.CreateNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		return fmt.Errorf("error while creating namespace: %v", err)
+	}
+	if err := npm.framework.ApplyTemplatedManifests(clusterRoleBindingFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating clusterRoleBinding: %v", err)
+	}
+	if err := npm.framework.ApplyTemplatedManifests(crdManifestFilePath, npm.templateMapping); err != nil {
+		return fmt.Errorf("error while creating CRD: %v", err)
+	}
+	return nil
+}
+
+func (npm *networkPerformanceMeasurement) cleanupCluster() {
+	if err := npm.framework.DeleteObject(crdGvk, "", customResourceDefinitionName); err != nil {
+		klog.Errorf("error while deleting CRD: %v", err)
+	}
+	if err := npm.framework.DeleteObject(rbacGvk, "", rbacName); err != nil {
+		klog.Errorf("error while deleting clusterRoleBinding: %v", err)
+	}
+	if err := client.DeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while deleting %s namespace: %v", netperfNamespace, err)
+	}
+	if err := client.WaitForDeleteNamespace(npm.k8sClient, netperfNamespace); err != nil {
+		klog.Errorf("error while waiting for %s namespace to be deleted: %v", netperfNamespace, err)
+	}
+}
+
+func (npm *networkPerformanceMeasurement) createAndWaitForWorkerPods() error {
+	// Create worker pods
+	var replicas = map[string]interface{}{"Replicas": npm.numberOfClients + npm.numberOfServers}
+	if err := npm.framework.ApplyTemplatedManifests(workerPodDeploymentManifestFilePath, replicas); err != nil {
+		return fmt.Errorf("error while creating worker pods :%v ", err)
+	}
+	// Wait for all worker pods to be ready
+	stopCh := make(chan struct{})
+	time.AfterFunc(podReadyTimeOut, func() {
+		close(stopCh)
+	})
+	options := &measurementutil.WaitForPodOptions{
+		Selector:            &measurementutil.ObjectSelector{Namespace: netperfNamespace},
+		DesiredPodCount:     npm.numberOfClients + npm.numberOfServers,
+		CallerName:          networkPerformanceMetricsName,
+		WaitForPodsInterval: 2 * time.Second,
+	}
+	return measurementutil.WaitForPods(npm.k8sClient, stopCh, options)
+}
+
+func (*networkPerformanceMeasurement) String() string {
+	return networkPerformanceMetricsName
+}
+
+func (npm *networkPerformanceMeasurement) storeWorkerPods() error {
+	pods, err := npm.k8sClient.CoreV1().Pods(netperfNamespace).List(context.TODO(), metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+	for _, pod := range pods.Items {
+		if pod.Status.PodIP == "" {
+			klog.Errorf("IP address not found for %s", pod.Name)
+			continue
+		}
+		podData := &workerPodData{podName: pod.Name, podIP: pod.Status.PodIP, workerNode: pod.Spec.NodeName}
```

Comment:
nit: please create newWorkerPodData(name, podIP, nodeName) helper function
and just switch to:
```
np.populateWorkerPodList(newWorkerPodData(pod.Name, pod.Status.PodIP, pod.Spec.NodeName))
```

Reply:
modified